imhamt/src/operation.rs

```diff
@@ -1,21 +1,32 @@
-#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+use thiserror::Error;
+
+#[derive(Debug, Error, Clone, Copy, PartialEq, Eq)]
 pub enum InsertError {
+    #[error("entry with the provided key already exists")]
```

Comment:
This is what was missing, I don't think the bounds on generic operations are needed though.